client/era/era_structs.go

```diff
@@ -0,0 +1,139 @@
+}
+
+// ListClustersResponse structs
+type ListClusterResponse struct {
+	ID                   *string         `json:"id,omitempty"`
+	Name                 *string         `json:"name,omitempty"`
+	Uniquename           *string         `json:"uniqueName,omitempty"`
+	Ipaddresses          []*string       `json:"ipAddresses,omitempty"`
+	Fqdns                interface{}     `json:"fqdns,omitempty"`
+	Nxclusteruuid        *string         `json:"nxClusterUUID,omitempty"`
+	Description          *string         `json:"description,omitempty"`
+	Cloudtype            *string         `json:"cloudType,omitempty"`
+	Datecreated          *string         `json:"dateCreated,omitempty"`
+	Datemodified         *string         `json:"dateModified,omitempty"`
+	Ownerid              *string         `json:"ownerId,omitempty"`
+	Status               *string         `json:"status,omitempty"`
+	Version              *string         `json:"version,omitempty"`
+	Hypervisortype       *string         `json:"hypervisorType,omitempty"`
+	Hypervisorversion    *string         `json:"hypervisorVersion,omitempty"`
+	Properties           []*Properties   `json:"properties,omitempty"`
+	Referencecount       int             `json:"referenceCount,omitempty"`
+	Username             interface{}     `json:"username,omitempty"`
+	Password             interface{}     `json:"password,omitempty"`
+	Cloudinfo            interface{}     `json:"cloudInfo,omitempty"`
+	Resourceconfig       *Resourceconfig `json:"resourceConfig,omitempty"`
+	Managementserverinfo interface{}     `json:"managementServerInfo,omitempty"`
+	Entitycounts         interface{}     `json:"entityCounts,omitempty"`
+	Healthy              bool            `json:"healthy,omitempty"`
+}
+
+type ClusterListResponse []ListClusterResponse
+
+type Resourceconfig struct {
+	Storagethresholdpercentage float64 `json:"storageThresholdPercentage,omitempty"`
+	Memorythresholdpercentage  float64 `json:"memoryThresholdPercentage,omitempty"`
+}
+
+// ListSLAResponse structs
+type ListSLAResponse struct {
+	ID                     *string `json:"id,omitempty"`
+	Name                   *string `json:"name,omitempty"`
+	Uniquename             *string `json:"uniqueName,omitempty"`
+	Description            *string `json:"description,omitempty"`
+	Ownerid                *string `json:"ownerId,omitempty"`
+	Datecreated            *string `json:"dateCreated,omitempty"`
+	Datemodified           *string `json:"dateModified,omitempty"`
+	CurrentActiveFrequency *string `json:"currentActiveFrequency,omitempty"`
+	Continuousretention    int     `json:"continuousRetention,omitempty"`
+	Dailyretention         int     `json:"dailyRetention,omitempty"`
+	Weeklyretention        int     `json:"weeklyRetention,omitempty"`
+	Monthlyretention       int     `json:"monthlyRetention,omitempty"`
+	Quarterlyretention     int     `json:"quarterlyRetention,omitempty"`
+	Yearlyretention        int     `json:"yearlyRetention,omitempty"`
```

Comment:
Please check if value 0 matters here or not. If yes, then int should be *int

Reply:
Check other places as well

Reply:
these are for datasources.